controllers/databaseclusterrestore_controller.go

```diff
@@ -599,23 +599,11 @@ func (r *DatabaseClusterRestoreReconciler) genPXCPitrRestoreSpec(
 	}
 
 	spec := &pxcv1.PITR{
-		BackupSource: &pxcv1.PXCBackupStatus{},
-		Type:         string(dataSource.PITR.Type),
-		Date:         dataSource.PITR.Date.Format(everestv1alpha1.DateFormatSpace),
-	}
-
-	switch backupStorage.Spec.Type {
-	case everestv1alpha1.BackupStorageTypeS3:
-		spec.BackupSource.S3 = &pxcv1.BackupStorageS3Spec{
-			CredentialsSecret: backupStorage.Spec.CredentialsSecretName,
-			Region:            backupStorage.Spec.Region,
-			EndpointURL:       backupStorage.Spec.EndpointURL,
-			Bucket:            pitrBucketName(sourceDB, backupStorage.Spec.Bucket),
-		}
-		//nolint:godox
-		// TODO: add support for Azure.
-	default:
-		return nil, fmt.Errorf("unsupported backup storage type %s for %s", backupStorage.Spec.Type, backupStorage.Name)
+		BackupSource: &pxcv1.PXCBackupStatus{
+			StorageName: pitrStorageName(*db.Spec.Backup.PITR.BackupStorageName),
+		},
+		Type: string(dataSource.PITR.Type),
+		Date: dataSource.PITR.Date.Format(everestv1alpha1.DateFormatSpace),
```

Comment:
Does restoring to a new cluster still work?

Reply:
The problem is that when restoring to a new cluster, it needs to use the same bucket as the source cluster..
With this change, it will use the bucket path that is configured for the new cluster, which won't work 🤔

Reply:
I see, a good point, thanks. We need to configure the bucket explicitly, but for that to use the storage that is configured for pitr but not for the backup

Reply:
I'll update it

Reply:
I don't think so. We still need to know the source database.
The restore PITR section needs to use the same bucket that is defined in the source database cluster in the PITR configuration. 

Updated the codebase, tested for both restoring to the same and to the new cluster.